### ISSUE TITLE
Gas calculated incorrectly for built-in contracts

### ISSUE THREAD

**Issue by chriseth** [ethereum/cpp-ethereum#1465]
The following contract does not work because the gas required for ripemd160 and sha256 is not calculated correctly.
```
contract c {
	
function testCryptographicFunctions() returns (bool res)							
{
    res = true;
    if (sha3('teststring') != 0x43c4b4524adb81e4e9a5c4648a98e9d320e3908ac5b6c889144b642cd08ae16d)		
        return false;

    if (sha256('teststring') != 0x3c8727e019a42b444667a587b6001251becadabbb36bfed8087a92c18882d111)		
        return false;											

    if (ripemd160('teststring') != 0xcd566972b5e50104011a92b59fa8e0b1234851ae)				
        return false;
}
	
}
```

**Comment by chriseth** [comment]
Might be fixed by #1546.

**Comment by chriseth** [comment]
Gas problem is fixed, but a new issue concerning byte alignment for `ripemd160` turned up.

**Comment by chriseth** [comment]
Related bug fixed in #1685.
